### PR TITLE
Added logger client with kwargs

### PR DESCRIPTION
**Pull request by liranbg**
Added a logging client to allow logging to either log to stdout/stream using human/json formatters.
Formatter can be easily changed by changing config from `human` to `json` (using envvar or hardcoded)

* Note that `logger` is still being imported from `mlrun.utils` and should be TO-DOed for later.

### CODE REVIEW

File: mlrun/api/main.py [hunk 1]
```diff
@@ -20,7 +20,7 @@
 
 @app.on_event("startup")
 async def startup_event():
-    logger.info("configuration dump\n%s", config.dump_yaml())
+    logger.info_with("configuration dump", dumped_config=config.dump_yaml())
```
Comment (Hedingber) [inline mlrun/api/main.py]:
```suggestion
    logger.info("configuration dump", dumped_config=config.dump_yaml())
```

File: mlrun/clients/logger.py [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+
+    # resolve formatter
+    formatter = _resolve_formatter(LoggerFormatterEnum(formatter.lower()))
+
+    # set handler
+    logger_instance.set_handler("default", stream or sys.stdout, formatter)
+
+    # disable event propagation to higher classes
+    # TODO: Why?!
+    logger_instance.propagate = False
```
Comment (Hedingber) [inline mlrun/clients/logger.py:133]: Why ? 😬

File: tests/clients/logger/test_logger.py [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+@pytest.fixture(scope="session")
+def stream() -> Generator:
+    io_stream = StringIO()
+    try:
+        yield io_stream
+    finally:
+        io_stream.close()
+
+
+@pytest.fixture(params=list(LoggerFormatterEnum.__members__))
```
Comment (Hedingber) [inline tests/clients/logger/test_logger.py:18]: I don't think the `.__members__` is needed

File: tests/clients/logger/test_logger.py [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    test_logger.debug("special_arg %s", "special_arg_value", name="special_kwarg")
+    assert "special_arg" in stream.getvalue()
+    assert "special_arg_value" in stream.getvalue()
+    assert "special_kwarg" in stream.getvalue()
+
+
+def test_with_kwargs(test_logger: Logger, stream: StringIO):
+    test_logger.debug("special_arg %s", name="special_kwarg")
+    assert "special_arg %s" in stream.getvalue()
+    assert "special_kwarg" in stream.getvalue()
```
Comment (Hedingber) [inline tests/clients/logger/test_logger.py:44]: Please verify that the `name` (maybe rename to `special_kwarg_key` or something) is in the log as well in both tests

File: mlrun/utils.py [hunk 5]
```diff
@@ -15,12 +15,12 @@
...
 import hashlib
 import json
-import logging
+import sys
 import pathlib
 import re
 from datetime import datetime, timezone
 from os import path, environ
-from sys import stdout
+from .clients.logger import create_logger
```
Comment (Hedingber) [inline mlrun/utils.py:23]: I don't feel like new `clients` module is needed
I suggest to change `utils` to a module which will have `logger.py` file and `helpers.py` file with the rest of the stuff, everything will be imported to the module's `__init__.py` file to keep BC (clients that do `from mlrun.utils import X` won't break)
WDYT ?

File: tests/clients/logger/test_logger.py (hunk 3 above)
Comment (liranbg) [reply tests/clients/logger/test_logger.py:18]: it...is

File: mlrun/utils.py (hunk 5 above)
Comment (liranbg) [reply mlrun/utils.py]: good idea, done

File: mlrun/clients/logger.py (hunk 2 above)
Comment (liranbg) [reply mlrun/clients/logger.py]: Was there before, I was wondering why? if no reason / no clue, I will remove

**Pull request merged**